packages/shorebird_cli/lib/src/commands/apps/list_apps_command.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:mason_logger/mason_logger.dart';
+import 'package:shorebird_cli/src/command.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+/// {@template list_apps_command}
+///
+/// `shorebird apps list`
+/// List all Shorebird apps.
+/// {@endtemplate}
+class ListAppsCommand extends ShorebirdCommand {
+  /// {@macro list_apps_command}
+  ListAppsCommand({
+    required super.logger,
+    super.buildCodePushClient,
+    super.auth,
+  });
+
+  @override
+  String get description => 'List all Shorebird apps.';
+
+  @override
+  String get name => 'list';
+
+  @override
+  List<String> get aliases => ['ls'];
+
+  @override
+  Future<int>? run() async {
+    final session = auth.currentSession;
+    if (session == null) {
+      logger.err('You must be logged in to publish.');
```

Comment:
"publish"

Reply:
This also feels like "middleware" at some level.  Some sort of bool on ShorebirdCommand constructor?

Reply:
Good catch will fix now